azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -220,28 +220,45 @@ public static void alertUserOnFirstError(final ExecutableFlow flow,
    */
   public static void failEverything(final ExecutableFlow exFlow) {
     final long time = System.currentTimeMillis();
-    for (final ExecutableNode node : exFlow.getExecutableNodes()) {
-      switch (node.getStatus()) {
-        case SUCCEEDED:
-        case FAILED:
-        case KILLED:
-        case SKIPPED:
-        case DISABLED:
-          continue;
-          // case UNKNOWN:
-        case READY:
-          node.setStatus(Status.KILLING);
-          break;
-        default:
-          node.setStatus(Status.FAILED);
-          break;
+    final LinkedList<ExecutableNode> queue = new LinkedList<>();
```

Comment:
Where are you using this queue? In this while loop, you have prepared it. Then traversing it and marking status is missing.

Reply:
This is incorrect way of declaring it. Always use the base class, in this case, List

Reply:
@jakhani the entire logic to mark the status is inside the while loop of the BFS.

Reply:
@djaiswal83 changed to Queue.
Discussed with @jakhani offline, the logic is correct.